### PR TITLE
CC-50: Add tests of partitioning by field value and time.

### PR DESCRIPTION
**Pull request by ewencp**

### CODE REVIEW

**Comment by ewencp** [comment]
@guozhangwang @gwenshap @Ishiihara whoever can get to this first... The test for time based partitioning is kind of weak because it's hard to create assertions for.

**Comment by ewencp** [comment]
@Ishiihara This got lost in the shuffle of the last release. Can you take a look so we can get this merged before we start with more bug fixes & features on the HDFS connector?

File: muckrake/tests/connect_hdfs_test.py [hunk 1]
```diff
@@ -144,6 +146,19 @@ def parse_partition_filename(path):
         extension = elems[4]
         return topic, partition, start_offset, end_offset, extension
 
+    @staticmethod
+    def parse_path_partition(path):
+        """
+        Get the directory that the given file has been categorized under for data partitioning purposes. By default,
+        partitioning is done by Kafka partition, so this will be of the form "partition=n". However, other groupings are
+        possible, such as field based partitioning (e.g. "field=value") and time-based (e.g. "FIXME"). Note that since the
```
Comment (Ishiihara) [inline muckrake/tests/connect_hdfs_test.py]: Should FIXME be fixed?

File: muckrake/tests/connect_hdfs_test.py [hunk 2]
```diff
@@ -300,10 +334,71 @@ def verify_no_compatibility(self):
...
+        self.connect.stop()
+        # Validation is limited. We should see contiguous ranges just like we do in default partitioning, although the
+        # layout will be different. Validate the Avro contents, then do some basic sanity checks on the datetimes in the
+        # output file paths
+        self.verify_avro(partitioning='time', partition='')
+        timestamps = []
+        for filename in self.ls(self.topic):
+            datetime_path = self.parse_path_partition(filename)
+            datetime_parts = [int(part) for part in datetime_path.split('/') if part]
+            # Can get subdirectories without data, i.e. /year/month/day, but not /year/month/day/hour/minute/second
```
Comment (Ishiihara) [inline muckrake/tests/connect_hdfs_test.py]: When can this happen?

**Comment by Ishiihara** [comment]
@ewencp Verified both tests. Left some minor comments, otherwise LGTM.

File: muckrake/tests/connect_hdfs_test.py (hunk 2 above)
Comment (ewencp) [inline muckrake/tests/connect_hdfs_test.py]: It is a recursive `ls` and that lists directories as well. I'll clarify the comment there -- it currently sounds like there are directories without data at all, but I actually just meant you get directories that don't have data directly in them, only in deeper subdirectories.

File: muckrake/tests/connect_hdfs_test.py (hunk 1 above)
Comment (ewencp) [inline muckrake/tests/connect_hdfs_test.py]: Haha, good catch. I think I wasn't sure what the format was going to be yet.

**Pull request merged**